src/managers/queue-manager.js

```diff
@@ -108,17 +70,21 @@ function handleMessage(message) {
 
 export async function pollMessageQueue() {
   if (getUserToken()) {
-    var response = await getUserQueue();
-    if (response && (response.status === 200 || response.status === 204)) {
-      log(`Message queue checked for user ${getUserToken()}, ${response.data.length} messages found.`);
-      if (response.data.length > 0) {
-        messages = response.data;
-        checkMessageQueue();
+    if (Gist.isDocumentVisible) {
+      var response = await getUserQueue();
+      if (response && (response.status === 200 || response.status === 204)) {
+        log(`Message queue checked for user ${getUserToken()}, ${response.data.length} messages found.`);
+        if (response.data.length > 0) {
+          messages = response.data;
+          checkMessageQueue();
+        } else {
+          log(`No messages for user token.`);    
+        }
       } else {
-        log(`No messages for user token.`);    
+        log(`There was an error while checking message queue.`);
       }
     } else {
```

Comment:
As a newbie to the gist-web repo, I would appreciate a bit more inline documentation when possible (and when the evaluation is not trivial) for the possible branches, like: 

_// If the condition evaluates to false, that means that the browser DOM is not visible in the user screen_
(Just as an example. I am actually not sure if it is correct, lol)

Nitpick: Also, I am personally not a huge fan of having multiple elses, I believe they could complicate how the program flow is understood. In general, when possible, I try use ifs and then returns instead. 
Maybe there's a bit of room for simplification or conditions swap here?

Reply:
Definitely, its something we can improve. In this case, we're only breaking them further for logging.

But yeah, looking at it now it does look intimidating 😂